mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -404,6 +429,30 @@ class RFormulaModel private[feature](
       s"Label column already exists and is not of type ${NumericType.simpleString}.")
   }
 
+  private def foldExprs(dataframe: DataFrame)(f: (DataFrame, String) => DataFrame): DataFrame =
+    resolvedFormula.evalExprs.foldLeft(dataframe)(f)
+
+  private def transformSelectExprs(dataframe: DataFrame): DataFrame = foldExprs(dataframe) {
+    case(df, colname) => df.withColumn(colname, expr(colname))
+  }
+
+  private def transformDropExprs(dataframe: DataFrame): DataFrame = foldExprs(dataframe) {
+    case(df, colname) => df.drop(col(s"`$colname`"))
+  }
+
+  private def transformSelectExprsSchema(schema: StructType): StructType = {
+    val spark = SparkSession.builder().getOrCreate()
+    val dummyRDD = spark.sparkContext.parallelize(Seq(Row.empty))
```

Comment:
should it use http://spark.apache.org/docs/latest/api/scala/index.html#org.apache.spark.SparkContext@emptyRDD[T](implicitevidence$8:scala.reflect.ClassTag[T]):org.apache.spark.rdd.RDD[T]